### PR TITLE
New serverless pattern - adobe-eventbridge-lambda

### PR DESCRIPTION
**Pull request by ragibmahsan**
*Issue #, if available:* #2326

*Description of changes:* New pattern submission for Adobe I/O Events to Lambda and CloudWatch

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by cornelcroi** [comment]
Hi,

There is a broken image link in the readme for the architecture image.
Can you update the PR?

Thanks.

**Comment by ragibmahsan** [comment]
Updated!

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/README.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+# Adobe Amazon EventBridge Partner Integration to AWS Lambda
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/README.md:1]:
```suggestion
# Send Adobe I/O Events to AWS Lambda through integration with Amazon EventBridge
```

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+This pattern demonstrates how to use the Adobe I/O Events Amazon EventBridge Integration and AWS Lambda to process/route events from Adobe. This pattern is leveraging the Adobe I/O Evenets Amazon EventBridge integration to send order events from the customer's Adobe account to their AWS account, via an Amazon EventBridge Partner event bus. Once the Adobe I/O events are in the customer's account, an Amazon EventBridge rule routes updated product events to a downstream Lambda function. In production cases, the Lambda function could transform the event, send it to a downstream application, archive it in a warehouse service, or send a notification email to a customer using SES. Amazon CloudWatch Log Groups are provisioned for debugging and auditing. This pattern deploys two EventBridge rules, one Lambda function, and two CloudWatch Log Groups.
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/README.md:3]:
```suggestion
This pattern demonstrates how to use the Adobe I/O Events integration with Amazon EventBridge to process/route events from Adobe with AWS Lambda. This pattern is leveraging the Adobe I/O Events integration with Amazon EventBridge to send order events from the customer's Adobe account to their AWS account, via an Amazon EventBridge Partner event bus. Once the Adobe I/O events are in the customer's account, an Amazon EventBridge rule routes updated product events to a downstream Lambda function. In a real-world use case, the Lambda function could transform the event, send it to a downstream application, archive it in a warehouse service, or send a notification email to a customer using SES. Amazon CloudWatch log groups are provisioned for debugging and auditing. This pattern deploys two EventBridge rules, one Lambda function, and two CloudWatch log groups.
```

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [AWS CDK CLI](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html) (AWS CDK) installed
+* [Create an Adobe Developer Account](https://www.adobe.com/sign/developer-form.html) if you do not already have one and log in.
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/README.md:15]: Please briefly mention if it is free and if so, under which conditions

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+## How it works
+
+This service interaction uses an existing Adobe SaaS integration in the customer's AWS account. If you do not have the Adobe SaaS integration set up in your AWS account, please set it up before deploying this pattern. View the [integration on Adobe's developer site](https://developer.adobe.com/events/docs/guides/amazon_eventbridge/).
+
+This pattern demonstrates how to:
+1. Write EventBridge rules that match Adobe's event pattern
+2. Send events from Adobe's EventBridge integration to Amazon CloudWatch for logging and debugging
+3. Transform Adobe I/O events using AWS Lambda and allows you to connect to other services going forward
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/README.md:48]:
```suggestion
3. Transform Adobe I/O events using AWS Lambda and how to connect to other services 
```

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json [hunk 5]
```diff
@@ -0,0 +1,90 @@
+{
+    "title": "Adobe EventBridge Integration to AWS Lambda",
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json:2]:
```suggestion
    "title": "Adobe I/O Events to AWS Lambda with Amazon EventBridge",
```

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    "description": "Use the Amazon EventBridge Adobe I/O Events integration to process/route order events with AWS Lambda, and notify users with Amazon SES or store the data elsewhere.",
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json:3]: The description can have at most 150 characters, please reword it to fit the limit

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    "language": "Python",
+    "level": "200",
+    "framework": "CDK",
+    "introBox": {
+      "headline": "How it works",
+      "text": [
+        "This sample project demonstrates how to use the Adobe I/O Events Amazon EventBridge Integration and AWS Lambda to process/route events from Adobe. This pattern is leveraging the Adobe I/O Evenets Amazon EventBridge integration to send order events from the customer's Adobe account to their AWS account, via an Amazon EventBridge Partner event bus. Once the Adobe I/O events are in the customer's account, an Amazon EventBridge rule routes updated product events to a downstream Lambda function.",
+        "Amazon CloudWatch Log Groups are provisioned for debugging and auditing.",
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json:11]:
```suggestion
        "Amazon CloudWatch log groups are provisioned for debugging and auditing.",
```

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        ]
+    },
+    "testing": {
+        "text": [
+            "See the GitHub repo for detailed testing instructions."
+        ]
+    },
+    "cleanup": {
+        "text": [
+            "cdk destroy"
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json:51]:
```suggestion
            "<code>cdk destroy</code>"
```

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+            "y": 50,
+            "service": "lambda",
+            "label": "AWS Lambda"
+        },
+        "icon3": {
+            "x": 80,
+            "y": 50,
+            "service": "cloudwatch",
+            "label": "Amazon CloudWatch Logs"
+        },
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json:80]: Does this need to be included in the diagram?

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/app.py [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    'service-role/AWSLambdaBasicExecutionRole')
+            ]
+        )
+
+        # A Lambda function to consume and process specific events
+        adobe_process_order_events_lambda = _lambda.Function(
+            self,
+            id='AdobeProcessOrderEventsLambda',
+            runtime=_lambda.Runtime.PYTHON_3_8,
```
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/app.py:59]: Python 3.8 is going to be deprecated soon. Please use a more up-to-date runtime

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/example-pattern.json
Comment (bfreiberg) [inline adobe-eventbridge-lambda/adobe-eventbridge-lambda/example-pattern.json:1]: Please remove this file if you already have the `adobe-eventbridge-lambda.json`

File: adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json (hunk 9 above)
Comment (ragibmahsan) [reply adobe-eventbridge-lambda/adobe-eventbridge-lambda/adobe-eventbridge-lambda.json:80]: I believe it does for quick overview of the architecture

**Comment by bfreiberg** [comment]
Please note that the description still exceeds the 150 character limit
